src/priority-queue.ts

```diff
@@ -0,0 +1,262 @@
+import {ArmorCollection} from './collection';
+import {ArmorCollectionSelector} from './selector';
+
+/**
+ * decide what to do when no comparator is passed
+ */
+
+export interface ArmorPriorityQueueOptions<T> {
+	elements?: Array<T>;
+	comparator?: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+}
+
+const defaultComparator = function (nodeOneIndex: number, nodeTwoIndex: number) {
+	if (typeof nodeOneIndex !== 'number') {
+		return false;
+	}
+	if (typeof nodeTwoIndex !== 'number') {
+		return false;
+	}
+
+	return JSON.stringify(this.elements[nodeOneIndex]) < JSON.stringify(this.elements[nodeTwoIndex]);
+};
+
+export class ArmorPriorityQueue<T> implements ArmorCollection<T> {
+	public elements: T[];
+	public comparator: (nodeOneIndex: number, nodeTwoIndex: number) => boolean;
+
+	constructor(options: ArmorPriorityQueueOptions<T> = {}) {
```

Comment:
The options object provides a great way for users to optionally add parameters that modify behavior without requiring it. However, it's not ideal for required parameters unless we need to pass in a large set of required properties. General best practice (with exceptions) is to place required parameters in the constructor, with optional options last.

- [ ] Remove comparator from options object.

- [ ] Add required comparator as first constructor argument.